Test failure real
```
FAIL: test_create_closed_child_case_and_close_parent_in_same_form (casexml.apps.phone.tests.test_sync_mode.SyncTokenUpdateTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "corehq/ex-submodules/casexml/apps/phone/tests/restore_test_utils.py", line 45, in inner
    return helper(*args, **kwargs)
  File "corehq/ex-submodules/casexml/apps/phone/tests/restore_test_utils.py", line 32, in __call__
    call_with_settings(fn_with_pre_and_post, run_config.settings, args, kwargs)
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/tzmigration/test_utils.py", line 28, in call_with_settings
    fn(*args, **kwargs)
  File "corehq/ex-submodules/casexml/apps/phone/tests/restore_test_utils.py", line 28, in fn_with_pre_and_post
    self.fn(*args, **kwargs)
  File "corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py", line 665, in test_create_closed_child_case_and_close_parent_in_same_form
    self._testUpdate(self.sync_log._id, {}, {})
  File "corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py", line 128, in _testUpdate
    self.assertEqual(len(dependent_case_id_map), len(sync_log.dependent_cases_on_phone))
AssertionError: 0 != 1
```

--- a/corehq/ex-submodules/casexml/apps/phone/models.py
+++ b/corehq/ex-submodules/casexml/apps/phone/models.py
@@ -642,8 +642,6 @@ class SimplifiedSyncLog(AbstractSyncLog):
         logger.debug('case ids before update: {}'.format(', '.join(self.case_ids_on_phone)))
         logger.debug('dependent case ids before update: {}'.format(', '.join(self.dependent_case_ids_on_phone)))
         logger.debug('index tree before update: {}'.format(self.index_tree))
-        skipped = set()
-        to_prune = set()
 
         class CaseUpdate(object):
             def __init__(self, case_id):

--- a/corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py
+++ b/corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py
@@ -638,7 +638,6 @@ class SyncTokenUpdateTest(SyncBaseTest):
         # they should both be gone
         self._testUpdate(self.sync_log._id, {}, {})
 
-    @run_with_all_restore_configs
     def test_create_closed_child_case_and_close_parent_in_same_form(self):
         # create the parent
         parent_id = self.factory.create_case()._id

--- a/corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py
+++ b/corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py
@@ -1,4 +1,5 @@
 import uuid
+from xml.etree import ElementTree
 from couchdbkit import ResourceNotFound
 from django.test.utils import override_settings
 from django.test import TestCase
@@ -618,7 +619,6 @@ class SyncTokenUpdateTest(SyncBaseTest):
         parent_id = self.factory.create_case()._id
         # create an irrelevent child and close the parent
         child_id = uuid.uuid4().hex
-        child_id = 'child'
         self.factory.create_or_update_cases([
             CaseStructure(
                 case_id=child_id,
@@ -653,6 +653,24 @@ class SyncTokenUpdateTest(SyncBaseTest):
                 attrs={'owner_id': 'irrelevant', 'close': True},
             )
         )
+
+    @run_with_all_restore_configs
+    def test_index_after_close(self):
+        parent_id = self.factory.create_case()._id
+        case_id = uuid.uuid4().hex
+        case_xml = self.factory.get_case_block(case_id, create=True, close=True)
+        # hackily insert an <index> block after the close
+        index_wrapper = ElementTree.Element('index')
+        index_elem = ElementTree.Element('parent')
+        index_elem.set('case_type', 'test')
+        index_elem.set('relationship', 'child')
+        index_elem.text = parent_id
+        index_wrapper.append(index_elem)
+        case_xml.append(index_wrapper)
+        self.factory.post_case_blocks([case_xml])
+        sync_log = get_properly_wrapped_sync_log(self.sync_log._id)
+        # before this test was written, the case stayed on the sync log even though it was closed
+        self.assertFalse(sync_log.phone_is_holding_case(case_id))
 
 
 class ChangingOwnershipTest(SyncBaseTest):

